PostWebApllication/src/ListOfPosts.js

```diff
@@ -0,0 +1,242 @@
+export class ListOfPosts extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            addposts: [],
+            modalTitle: "",
+            PostID: '',
+            PostTittle: "",
+            DescriptionOfPost: "",
+            CreatedDate: "",
+            UpdatedDate: "",
+            LikesCount: "",
+            HeartCount: "",
+            modalPostId: null
+        }
+    }
+    refreshList() {
+        debugger;
+        fetch(variables.API_URL + 'AddPost/GetPosts')
+            .then(response => response.json())
+            .then(data => {
+                this.setState({ addposts: data });
+            });
+    }
+    debugger;
+    componentDidMount() {
+        this.refreshList();
+    }
+    changePostTittle = (e) => {
+        this.setState({ PostTittle: e.target.value });
+    }
+    changeDescriptionOfPost = (e) => {
+        this.setState({ DescriptionOfPost: e.target.value });
+    }
+
+    addClick() {
+        debugger;
+        this.setState({
+            modalTitle: "Add Post",
+            PostId: '',
+            PostTittle: "",
+            DescriptionOfPost: "",
+            CreatedDate: "",
+            UpdatedDate: "",
+            LikesCount: "",
+            HeartCount: ""
+        });
+    }
+    editClick(a1) {
+        debugger;
+        this.setState({
+            modalTitle: "Edit Post",
+            PostID: a1.PostID,
+            PostTittle: a1.PostTittle,
+            DescriptionOfPost: a1.DescriptionOfPost,
+        });
+    }
+    debugger;
+    updateClick() {
+        debugger;
+        fetch(variables.API_URL + 'AddPost/UpdatePost', {
+            method: 'PUT',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                PostID: this.state.PostID,
+                PostTittle: this.state.PostTittle,
+                DescriptionOfPost: this.state.DescriptionOfPost,
+            })
+        })
+            .then(res => res.json())
+            .then((result) => {
+                alert(result);
+                this.refreshList();
+            }, (error) => {
+                alert('Failed');
+            })
+    }
+    debugger;
+    createClick() {
+        debugger;
+        fetch(variables.API_URL + 'AddPost/CreatePost', {
+
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                PostTittle: this.state.PostTittle,
+                DescriptionOfPost: this.state.DescriptionOfPost,
+            })
+        })
+            .then(res => res.json())
+            .then((result) => {
+                alert(result);
+                this.refreshList();
+            }, (error) => {
+                alert('Failed');
+            })
+    }
+    LikeClick(row) {
```

Comment:
Always use the naming convention for all your code. For example:

1. Variable and function names are written as **camelCase**
2. Global variables are written in **UPPERCASE** 
3. Constants (like PI) written in **UPPERCASE**

Reply:
i have used  naming convention where i required